File: webapp/frontend/src/api/types/health.ts
import { BaseResponse } from "./common";

export interface HealthResponse extends BaseResponse {
  status: string;
  service: string;
  version: string;
  time_utc: string;
}

export interface HealthMetricsResponse extends BaseResponse {
  api_latency_ms: number;
  error_rate_pct: number;
  uptime_pct: number;
  active_jobs: number;
  drift_detected: boolean;
  models_in_prod: number;
}

File: webapp/backend/app/schemas/health.py
from __future__ import annotations

from pydantic import BaseModel

from .common import BaseResponse


class HealthResponse(BaseResponse):
    status: str
    service: str
    version: str
    time_utc: str


class HealthMetricsResponse(BaseResponse):
    api_latency_ms: float
    error_rate_pct: float
    uptime_pct: float
    active_jobs: int
    drift_detected: bool
    models_in_prod: int

